SimpleIT.simpleExchangeMethods fails in native mode with Mandrel 23.1 JDK 21

### Bug description

Bit confused as to what's going on here. `SimpleIT` when run in isolation within `integration-test-groups/foundation/core-languages` passes. When the test is run 'grouped' in `integration-tests/foundation-grouped` it fails with:

```
2023-10-19 11:17:05,014 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /core-languages/route/simpleExchangeMethods/String failed, error id: 4194163d-7dce-4450-8798-d31e732c4d0b-1: org.jboss.resteasy.spi.UnhandledException: org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively invoke method public abstract org.apache.camel.Message org.apache.camel.Exchange.getMessage() without it being registered for runtime reflection. Add public abstract org.apache.camel.Message org.apache.camel.Exchange.getMessage() to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:107)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:344)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:205)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:452)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base@21/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:211)
Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively invoke method public abstract org.apache.camel.Message org.apache.camel.Exchange.getMessage() without it being registered for runtime reflection. Add public abstract org.apache.camel.Message org.apache.camel.Exchange.getMessage() to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.forQueriedOnlyExecutable(MissingReflectionRegistrationUtils.java:72)
        at java.base@21/java.lang.reflect.Method.acquireMethodAccessor(Method.java:77)
        at java.base@21/java.lang.reflect.Method.invoke(Method.java:577)
        at org.apache.camel.support.ObjectHelper.invokeMethodSafe(ObjectHelper.java:385)
        at org.apache.camel.component.bean.MethodInfo.invoke(MethodInfo.java:493)
        at org.apache.camel.component.bean.MethodInfo$1.doProceed(MethodInfo.java:315)
        at org.apache.camel.component.bean.MethodInfo$1.proceed(MethodInfo.java:285)
        at org.apache.camel.component.bean.AbstractBeanProcessor.useMethodInvocation(AbstractBeanProcessor.java:122)
        at org.apache.camel.component.bean.AbstractBeanProcessor.process(AbstractBeanProcessor.java:88)
        at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:81)
        at org.apache.camel.support.AsyncProcessorSupport.process(AsyncProcessorSupport.java:32)
        at org.apache.camel.language.bean.BeanExpression.invokeBean(BeanExpression.java:364)
        at org.apache.camel.language.bean.BeanExpression.invokeOgnlMethod(BeanExpression.java:466)
        at org.apache.camel.language.bean.BeanExpression.evaluate(BeanExpression.java:216)
        at org.apache.camel.language.bean.BeanExpression.evaluate(BeanExpression.java:230)
        at org.apache.camel.language.simple.SimpleExpressionBuilder$29.evaluate(SimpleExpressionBuilder.java:761)
        at org.apache.camel.support.ExpressionAdapter.evaluate(ExpressionAdapter.java:45)
        at org.apache.camel.support.builder.ExpressionBuilder$55.evaluate(ExpressionBuilder.java:1709)
        at org.apache.camel.support.ExpressionAdapter.evaluate(ExpressionAdapter.java:45)
        at org.apache.camel.processor.TransformProcessor.process(TransformProcessor.java:47)
        at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.doRun(RedeliveryErrorHandler.java:824)
        at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$RedeliveryTask.run(RedeliveryErrorHandler.java:732)
        at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.executeReactiveWork(DefaultReactiveExecutor.java:196)
        at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:164)
        at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:59)
        at org.apache.camel.processor.Pipeline.process(Pipeline.java:163)
        at org.apache.camel.impl.engine.CamelInternalProcessor.process(CamelInternalProcessor.java:383)
        at org.apache.camel.component.direct.DirectProducer.process(DirectProducer.java:96)
        at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:212)
        at org.apache.camel.impl.engine.SharedCamelInternalProcessor$1.process(SharedCamelInternalProcessor.java:109)
        at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:81)
        at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:106)
        at org.apache.camel.support.cache.DefaultProducerCache.send(DefaultProducerCache.java:180)
        at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:176)
        at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:172)
        at org.apache.camel.impl.engine.DefaultProducerTemplate.requestBody(DefaultProducerTemplate.java:413)
        at org.apache.camel.quarkus.core.languages.it.CoreLanguagesResource.resource(CoreLanguagesResource.java:66)
        at org.apache.camel.quarkus.core.languages.it.CoreLanguagesResource_ClientProxy.resource(Unknown Source)
        at java.base@21/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:356)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:70)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
        ... 18 more
```

--- a/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelSerializationProcessor.java
+++ b/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelSerializationProcessor.java
@@ -31,8 +31,6 @@ import java.util.concurrent.ConcurrentSkipListSet;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
-import org.apache.camel.CamelExecutionException;
-import org.apache.camel.RuntimeCamelException;
 import org.apache.camel.quarkus.core.CamelConfig;
 import org.apache.camel.quarkus.core.deployment.spi.CamelSerializationBuildItem;
 import org.apache.camel.support.DefaultExchangeHolder;
@@ -69,9 +67,7 @@ public class CamelSerializationProcessor {
             Throwable.class.getName(),
 
             // Camel classes
-            CamelExecutionException.class.getName(),
             DefaultExchangeHolder.class.getName(),
-            RuntimeCamelException.class.getName(),
     };
 
     @BuildStep

--- a/integration-test-groups/foundation/core-languages/src/test/java/org/apache/camel/quarkus/core/languages/it/SimpleTest.java
+++ b/integration-test-groups/foundation/core-languages/src/test/java/org/apache/camel/quarkus/core/languages/it/SimpleTest.java
@@ -16,7 +16,6 @@
  */
 package org.apache.camel.quarkus.core.languages.it;
 
-import io.quarkus.test.junit.DisabledOnIntegrationTest;
 import io.quarkus.test.junit.QuarkusTest;
 import org.junit.jupiter.api.Test;
 
@@ -69,7 +68,6 @@ public class SimpleTest {
                 .body(is("BYTE_BUFFER"));
     }
 
-    @DisabledOnIntegrationTest("https://github.com/apache/camel-quarkus/issues/5444")
     @Test
     public void simpleExchangeMethods() {
         given().body("Joe").when().post("/core-languages/route/simpleExchangeMethods/String").then()

